test/object_store/conftest.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+
+import os
+import sys
+import logging
+import pytest
+import shutil
+import tempfile
+from dataclasses import dataclass
+
+# use minio_server
+sys.path.insert(1, sys.path[0] + '/../..')
+from test.pylib.minio_server import MinioServer
+from test.pylib.host_registry import HostRegistry
+from test.pylib.cql_repl import conftest
+
+hosts = HostRegistry()
+
+
+def pytest_addoption(parser):
+    conftest.pytest_addoption(parser)
+    # reserved for tests with real S3
+    s3_options = parser.getgroup("s3-server", description="S3 Server settings")
+    s3_options.addoption('--s3-server-address')
+    s3_options.addoption('--s3-server-port', type=int)
+    s3_options.addoption('--s3-server-bucket')
+
+
+@dataclass
+class S3_Server:
+    address: str
+    port: int
+    bucket_name: str
+
+    async def start(self):
+        pass
+
+    async def stop(self):
+        pass
+
+
+@pytest.fixture(scope="function")
+def ssl(request):
+    yield request.config.getoption('--ssl')
+
+
+@pytest.fixture(scope="function")
+def test_tempdir():
+    # with tempfile.TemporaryDirectory(prefix='scylla-test-') as tempdir:
+    tempdir = tempfile.mkdtemp(prefix='scylla-test-')
+    yield tempdir
+    with open(os.path.join(tempdir, 'log'), 'rb') as log:
+        shutil.copyfileobj(log, sys.stdout.buffer)
+
+@pytest.fixture(scope="function")
+async def tempdir_and_s3_server(pytestconfig, test_tempdir: str):
+
+    server = None
+    s3_server_address = pytestconfig.getoption('--s3-server-address')
+    s3_server_port = pytestconfig.getoption('--s3-server-port')
+    s3_server_bucket = pytestconfig.getoption('--s3-server-bucket')
+
+    default_address = os.environ.get('S3_SERVER_ADDRESS_FOR_TEST')
+    default_port = os.environ.get('S3_SERVER_PORT_FOR_TEST')
+    default_bucket = os.environ.get('S3_PUBLIC_BUCKET_FOR_TEST')
+
+    if s3_server_address:
+        server = S3_Server(s3_server_address,
+                           s3_server_port,
+                           s3_server_bucket)
+    elif default_address:
```

Comment:
How can this happen? The environment is only set by MinioServer which is started in the else branch of this elif

Reply:
there are three use cases of this test:

1. launched directly by pytest command line, with the parameters of `--s3-server-address` and friends. this is handled by the `if` case. and it has the highest priority
2. launched directly by pytest command line, without passing the parameters of `--s3-server-address` and friends. this is handled by the `else` case.
3. launched by `test.py`. so `test.py` prepares the minio_server, and sets the environmental variables. this is handled by the `elif` branch.

Reply:
Ah, I see. Not sure we need this flexibility, but the code is here so why not...